cpu/nrf52/vectors.c

```diff
@@ -96,8 +98,43 @@ WEAK_DEFAULT void isr_pwm3(void);
 extern void SWI2_EGU2_IRQHandler(void);
 #endif
 
+#if defined(CPU_MODEL_NRF52811XXAA)
 /* CPU specific interrupt vector table */
 ISR_VECTOR(1) const isr_t vector_cpu[CPU_IRQ_NUMOF] = {
+    isr_power_clock,       /* power_clock */
+    isr_radio,             /* radio */
+    isr_uart0,             /* uart0 */
+    isr_spi1_twi1,         /* spi1_twi1 */
+    isr_spi0,              /* spi 0 */
```

Comment:
Hm now I wonder - are you sure those are not just

```C
    isr_spi0_twi0,         /* spi0_twi0 */
    isr_spi1_twi1,         /* spi1_twi1 */
```

?

Reply:
No, I am not sure. I just copied it from the[ datasheet](https://www.nordicsemi.com/-/media/DocLib/Other/Product_Spec/nRF52811PSv10pdf.pdf). Don't know how it works in detail. But even though, it is wrong. I just looked into it again. It should be `spi1_twi0` Just compiled some examples for the nrf52811dk so far. Want to try a SPI example in the near future.

Reply:
@benpicco Would be great, if you can tell where these variables are coming from or what happens to them? Where do they get the addresses from?

Reply:
This is the interrupt vector table.
When an interrupt occurs the CPU uses the ID of the interrupt source as an index to decide which function to call to handle the interrupt.

So interrupts are enabled for the SPI peripheral, those functions will get called.
They are currently unused, but #14057 will change this.

Reply:
@benpicco Yes, I understood this. Thanks for the information that they are not being used. This explains why I could find it :D

Reply:
On which page of the data sheet is it?
I couldn't find it yet.

Reply:
Ah, tricky

> There is a direct relationship between peripheral ID and base address. For example, a peripheral with base
address 0x40000000 is assigned ID=0, a peripheral with base address 0x40001000 is assigned ID=1, and a
peripheral with base address 0x4001F000 is assigned ID=31.

![image](https://user-images.githubusercontent.com/1301112/81674968-54d7a500-944e-11ea-968f-b63fcf7cc1ad.png)

Reply:
> Ah, tricky
> 
> > There is a direct relationship between peripheral ID and base address. For example, a peripheral with base
> > address 0x40000000 is assigned ID=0, a peripheral with base address 0x40001000 is assigned ID=1, and a
> > peripheral with base address 0x4001F000 is assigned ID=31.
> 
> ![image](https://user-images.githubusercontent.com/1301112/81674968-54d7a500-944e-11ea-968f-b63fcf7cc1ad.png)

Yep, exactly the part I messed up. It would make sense to have spi0 and twi0 together, but it isn't ^^
Yes, this is why I thought the order matters. I wasn't sure about the deprecated things so. I kept it like it currently is. But spi should be spim or spis. 
This is probably also a bit connected to #10587 then.